content/blog/2017/2017-02-07-declarative-maven-project.adoc

```diff
@@ -235,17 +238,55 @@ pipeline {
                 sh 'mvn -Dmaven.test.failure.ignore=true install' // <1>
             }
             post {
-                always {
+                success {
                     junit 'target/surefire-reports/**/*.xml' // <2>
                 }
             }
         }
     }
 }
-// Scripted Not Defined //
+// Scripted
+node {
+    checkout scm
+
+    String jdktool = tool name: "jdk8", type: 'hudson.model.JDK'
+    def mvnHome = tool name: 'mvn'
+
+    /* Set JAVA_HOME, and special PATH variables. */
+    List javaEnv = [
+        "PATH+MVN=${jdktool}/bin:${mvnHome}/bin",
+        "M2_HOME=${mvnHome}",
+        "JAVA_HOME=${jdktool}"
+    ]
+
+    withEnv(javaEnv) {
+    stage ('Initialize') {
+        sh '''
+            echo "PATH = ${PATH}"
+            echo "M2_HOME = ${M2_HOME}"
+        '''
+    }
+    stage ('Build') {
+        try {
+            sh 'mvn -Dmaven.test.failure.ignore=true install'
+        } catch (e) {
+            currentBuild.result = 'FAILURE'
+        }
```

Comment:
The `try`/`catch` should be unnecessary.  If Maven fails, then the result will automatically be marked as a failure, and the build will stop at this point.  In the Pipeline Stage View, it will be clear to see that the "Build" stage failed.

As it is just now, the build will continue the "Post" stage, successfully do nothing, and leave all stages marked as red in the Stage View, which isn't particularly helpful.